keps/sig-network/2004-topology-aware-subsetting/README.md

```diff
@@ -565,6 +583,165 @@ label. Each approach may group the endpoints in slightly different
 combinations of EndpointSlices, but it will still be just as easy to list all
 endpoints for a Service.
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: TopologyAwareSubsetting
+    - Components depending on the feature gate: kube-controller-manager
+  - [x] Other
+    - Describe the mechanism:
+      `endpointslice.kubernetes.io/subsetting` annotation on Service.
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+      No
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node?
+      No
+  - [x] Other
+    - Describe the mechanism:
+      Controller flag to change inferred value when annotation is not set.
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+      No
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node?
+      No
+
+* **Does enabling the feature change any default behavior?**
+  In most cases, this feature will be intentionally enabled on specific Services
+  with the annotation described above. This will only have an effect if the
+  feature gate is enabled on kube-controller-manager. Even then, this will just
+  represent a different way to group EndpointSlices and will not have any
+  significant effect unless the consumer of EndpointSlices is also configured to
+  support this feature (see KEP 2030). If all of those bits are enabled, traffic
+  will be more likely to be routed to endpoints within the same zone.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes. It can easily be disabled universally by turning off the feature gate or
+  on a specific Service by setting the annotation to `Disable`.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  EndpointSlices will be subset again resulting in changes to existing
+  EndpointSlices for Services that have this feature enabled.
+
+* **Are there any tests for feature enablement/disablement?**
+  This feature is not yet implemented but per-Service enablement/disablement is
+  covered in depth as part of the test plan.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  The biggest risk here is that EndpointSlices may be created for some but not
+  all zones. This could result in a service being unreachable from zones that do
+  not have corresponding EndpointSlices. Importantly this would only be an issue
+  if all EndpointSlices for a Service have the `for-zone` label and 1 or more
```

Comment:
Should we arrange kube-proxy to detect that there are no endpoints for "my" zone and fall back on another for-zone?  It would be a real change in how the watches were set up and would erase some of the optimization, but it offers some deeper defense against problems.

Reply:
Updated to mention that fallback behavior.